awx/sso/pipeline.py

```diff
@@ -229,3 +229,70 @@ def update_user_teams_by_saml_attr(backend, details, user=None, *args, **kwargs)
 
     if team_map.get('remove', True):
         [t.member_role.members.remove(user) for t in Team.objects.filter(Q(member_role__members=user) & ~Q(id__in=team_ids))]
+
+
+def _check_flag(user, flag, attributes, user_flags_settings):
+    new_flag = False
+    is_role_key = "is_%s_role" % (flag)
+    is_attr_key = "is_%s_attr" % (flag)
+    is_value_key = "is_%s_value" % (flag)
+
+    # Check to see if we are respecting a role and, if so, does our user have that role?
+    role_setting = user_flags_settings.get(is_role_key, None)
+    if role_setting:
+        # We do a 2 layer check here so that we don't spit out the else message if there is no role defined
+        if role_setting in attributes.get('Role', []):
+            logger.debug("User %s has %s role %s" % (user.username, flag, role_setting))
+            new_flag = True
+        else:
+            logger.debug("User %s is missing the %s role %s" % (user.username, flag, role_setting))
+
+    # Next, check to see if we are respecting an attribute; this will take prio over the role if its defined
+    attr_setting = user_flags_settings.get(is_attr_key, None)
+    if attr_setting and attributes.get(attr_setting, None):
+        # Do we have a required value for the attribute
+        if user_flags_settings.get(is_value_key, None):
+            # If so, check and see if the value of the attr matches the required value
+            attribute_value = attributes.get(attr_setting, None)
+            if isinstance(attribute_value, (list, tuple)):
+                attribute_value = attribute_value[0]
```

Comment:
Why is this sometimes a list, and why is the first element desirable?

Reply:
This should be a function of how the IdP is sending the data to the SP. It's possible for an IdP to send multiple values for attributes or a single attribute. The SAML adapter already used the prescience of just picking the first item if it was a list so I followed suit.